component/host.go

```diff
@@ -29,6 +29,10 @@ type Host interface {
 	// before Component.Shutdown() begins.
 	ReportFatalError(err error)
 
+	ReportStatus(report StatusReport)
```

Comment:
Is there an expectation that the Host should react in a specific way to the StatusReport besides propagating to the registered consumers?